Added custom notification script

domibarton opened the pull request:
Hi there!

As proposed in #435 I've added a new notification type which basically just executes a custom script. The script is executed with a __title__ and __msg__ CLI argument like this:

```bash
/path/to/the/script "TITLE" "MSG"
```

I tried to adopt the existing "code style" and tried to use your naming scheme. I saw that there is a _Notification Center_ notify which is shortened to `ncenter` in the code. So I decided to call it _Notification Script_ (verbose) and therefore `nscript` in the code. Hope that's OK!

Let me know if you're OK with it or if I've to change anything.

Cheers
Domi

Safihre commented:
I kind of like this more than I thought I would, since this would sort of excuse us from having to add any more notification services since those can be handled by scripts :smile:
On the other hand, the main notifications people care about (I think) will be job succes/failure, and those can be handled by regular end-of-job scripts. You would have to turn off 'Post-Process Only Verified Jobs' so it can also handle failed jobs.

But yes, some cleanup of the code would be welcome! growler.py kind of grew and grey over the years, even the filename reminds of that!

PS: Greetings from Zurich ;-)

shypike commented:
Thanks for your contribution.
Unfortunately, I'm too busy with 0.8.0 issues to review this now.
Also, we have a feature-freeze due to the pending 0.8.0 release.
It will of course be included in a release after 0.8.0

domibarton commented:
Awesome, thanks!

Safihre commented on sabnzbd/growler.py:
```diff
@@ -539,7 +545,57 @@ def send_pushbullet(title, msg, gtype, force=False, test=None):
+    if force:
+        prio = True
+
+    if prio:
```
You don't need any of all this prio stuff, that's only for Prowl/Pushbullet etc. that support notification priorities. 
Look for example at the one for Windows notifications, all you need to do is call the script directly:
```python
def send_windows(title, msg, gtype):
    if sabnzbd.WINTRAY:
        sabnzbd.WINTRAY.sendnotification(title, msg)
    return None
```

domibarton commented:
@Safihre Are you sure? Isn't that required to decide which kind of messages (e.g. startup/shutdown) should be sent? 

I know on Prowl / Pushover the user can select the priority (integer) and I don't need that. But have a look at Pushbullet. The user can enable/disable notifications (boolean) based on the message type (e.g. startup, download started, post processing, download complete, disk full etc.). 

I thought it would be nice if we do the same thing for notification scripts, so that the user can select which notifications should invoke the notification script. However, if you want to send all notifications independent of the message type to the notification script, then I can remove it :)

Cheers
Domi

Safihre commented:
I forgot a part to post, the ```check_classes``` (see line 112) function will filter that only the ones that are checked will be send:
```python
    # Windows
    if sabnzbd.WIN32 and sabnzbd.cfg.acenter_enable():
        if check_classes(gtype, 'acenter'):
            send_windows(title, msg, gtype)
```
It should have also been this way for Pushbullet I think, maybe just a mistake.

Safihre commented:
@shypike besides the Pushbullet not needing the whole prio-if-things, I also saw 2 errors:
https://github.com/sabnzbd/sabnzbd/blob/develop/sabnzbd/growler.py#L393
https://github.com/sabnzbd/sabnzbd/blob/develop/sabnzbd/growler.py#L506
```disk-full``` should be ```disk_full```, that's the way it's variable is called in the ```cfg```

Probably should have a ```set_prio``` function that can work almost identical to the way ```check_classes``` does it instead of all those if's.

domibarton commented:
Sorry I was busy the last couple of days.
However, I fixed the errors and rebased my branch, which unfortunately closed this PR ;)

So I opened another one. 
Second one for the Pushbullet fix is coming soon...

shypike commented:
Closed with ccba56e073049c3c092f2be179b0a56d0ce4e798
